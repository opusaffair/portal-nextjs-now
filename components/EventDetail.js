import { Query } from "react-apollo";
import gql from "graphql-tag";

export const EVENT_DETAIL_QUERY = gql`
  query eventDetail($slug: String!) {
    event(slug: $slug) {
      title
      image_url
      organizer_desc
    }
  }
`;

export default function EventDetail(props) {
  return (
    <Query query={EVENT_DETAIL_QUERY} variables={props}>
      {({ loading, error, data }) => {
        if (error) return <div>Errors</div>;
        if (loading) return <div>Loading</div>;
        const { event } = data;
        if (!event) return <div>No match</div>;
        return (
          <section>
            <h1>{event.title}</h1>
            <img src={`${event.image_url}?w=620&h=350&fit=crop&crop=faces`} />
            <div
              dangerouslySetInnerHTML={{
                __html: `${event.organizer_desc}`
              }}
            />
          </section>
        );
      }}
    </Query>
  );
}
