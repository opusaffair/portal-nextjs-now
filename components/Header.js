import React from "react";
import Link from "next/link";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Button from "@material-ui/core/Button";

const Header = () => {
  return (
    <AppBar postition="static" color="primary">
      <Toolbar>
        <Link href="/">
          <Button color="inherit">Demo Calendar Microsite</Button>
        </Link>

        <Link href="/upcoming">
          <Button color="inherit">Upcoming</Button>
        </Link>

        <Link href="/past">
          <Button color="inherit">Past</Button>
        </Link>

        <Link href="/about">
          <Button color="inherit">About</Button>
        </Link>
        <Link href="/submit">
          <Button color="inherit">Submit</Button>
        </Link>
      </Toolbar>
    </AppBar>
  );
};

export default Header;
