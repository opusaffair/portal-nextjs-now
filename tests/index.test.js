/* eslint-env jest */

import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import Header from "../components/Header";
import EventDetail from "../components/EventDetails";

describe("With Snapshot Testing", () => {
  it("Header renders", () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  //   it("EventDetail renders", () => {
  //     const component = renderer.create(
  //       <EventDetail slug="boston-youth-symphony-orchestra-la-boheme" />
  //     );
  //     const tree = component.toJSON();
  //     expect(tree).toMatchSnapshot();
  //   });
});
