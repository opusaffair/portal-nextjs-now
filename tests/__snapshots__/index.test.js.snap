// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Header renders 1`] = `
<header
  className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-2 MuiAppBar-positionFixed-3 MuiAppBar-colorPrimary-9 mui-fixed"
  postition="static"
>
  <div
    className="MuiToolbar-root-38 MuiToolbar-regular-40 MuiToolbar-gutters-39 Header-innerToolbar-1"
  >
    <button
      className="MuiButtonBase-root-69 MuiButton-root-43 MuiButton-text-45 MuiButton-flat-48 MuiButton-colorInherit-64 Logo-logo-42"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-44"
      >
        StagePage
      </span>
      <span
        className="MuiTouchRipple-root-72"
      />
    </button>
    <button
      className="MuiButtonBase-root-69 MuiButton-root-43 MuiButton-text-45 MuiButton-flat-48 MuiButton-colorInherit-64"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-44"
      >
        About
      </span>
      <span
        className="MuiTouchRipple-root-72"
      />
    </button>
  </div>
</header>
`;
